src/Compilers/VisualBasic/Portable/Parser/ParseExpression.vb

```diff
@@ -1627,7 +1627,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
 
             Debug.Assert(CurrentToken.Kind = SyntaxKind.FunctionKeyword OrElse
                          CurrentToken.Kind = SyntaxKind.SubKeyword,
-                         "ParseFunctionLambda called on wrong token.")
+                         NameOf(ParseFunctionOrSubLambdaHeader) & " called on wrong token.")
```

Comment:
@gafter I suspect (but can not confirm through the public repo history git and codeplex), that originally there could have been two separate methods. One of `Sub Lambda` and `Function Lambda`, which at later date was unified. The member name in the string may have been missed.